app/code/Magento/Inventory/Setup/Operation/CreateReservationTable.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Setup\Operation;
+
+use Magento\Framework\DB\Adapter\AdapterInterface;
+use Magento\Framework\DB\Ddl\Table;
+use Magento\Framework\Setup\SchemaSetupInterface;
+use Magento\Inventory\Model\ResourceModel\Reservation as ReservationResourceModel;
+use Magento\Inventory\Model\ResourceModel\Stock as StockResourceModel;
+use Magento\InventoryApi\Api\Data\ReservationInterface;
+use Magento\InventoryApi\Api\Data\StockInterface;
+
+class CreateReservationTable
+{
+    /**
+     * @param SchemaSetupInterface $setup
+     * @return void
+     */
+    public function execute(SchemaSetupInterface $setup)
+    {
+        $reservationTable = $this->createReservationTable($setup);
+
+        $setup->getConnection()->createTable($reservationTable);
+    }
+
+    /**
+     * @param SchemaSetupInterface $setup
+     * @return Table
+     */
+    private function createReservationTable(SchemaSetupInterface $setup)
+    {
+        $reservationTable = $setup->getTable(ReservationResourceModel::TABLE_NAME_RESERVATION);
+        $stockTable = $setup->getTable(StockResourceModel::TABLE_NAME_STOCK);
+
+        return $setup->getConnection()->newTable(
+            $reservationTable
+        )->setComment(
+            'Inventory Reservation Table'
+        )->addColumn(
+            ReservationInterface::RESERVATION_ID,
+            Table::TYPE_INTEGER,
+            null,
+            [
+                Table::OPTION_IDENTITY => true,
+                Table::OPTION_UNSIGNED => true,
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_PRIMARY => true,
+            ],
+            'Reservation ID'
+        )->addColumn(
+            ReservationInterface::STOCK_ID,
+            Table::TYPE_INTEGER,
+            null,
+            [
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_UNSIGNED => true,
+            ],
+            'Stock ID'
+        )->addColumn(
+            ReservationInterface::SKU,
+            Table::TYPE_TEXT,
+            64,
+            [
+                Table::OPTION_NULLABLE => false,
+            ],
+            'Sku'
+        )->addColumn(
+            ReservationInterface::QUANTITY,
+            Table::TYPE_DECIMAL,
+            null,
+            [
+                Table::OPTION_UNSIGNED => false,
+                Table::OPTION_NULLABLE => false,
+                Table::OPTION_DEFAULT => 0,
+            ],
+            'Quantity'
+        )->addColumn(
+            ReservationInterface::METADATA,
+            Table::TYPE_TEXT,
+            '64k',
```

Comment:
not sure that  255 symbols would be enough for us. 
But in current Magento implementation, framewrok detects by itself based on the provided length which MySQL data type should be used for storing data.

We definitely want that Magento created varchar data type, not the `mediumtext` one.
```php
                if ($length <= 255) {
                    $cType = $ddlType == Table::TYPE_TEXT ? 'varchar' : 'varbinary';
                    $cType = sprintf('%s(%d)', $cType, $length);
                } elseif ($length > 255 && $length <= 65536) {
                    $cType = $ddlType == Table::TYPE_TEXT ? 'text' : 'blob';
                } elseif ($length > 65536 && $length <= 16777216) {
                    $cType = $ddlType == Table::TYPE_TEXT ? 'mediumtext' : 'mediumblob';
                } else {
                    $cType = $ddlType == Table::TYPE_TEXT ? 'longtext' : 'longblob';
                }
```
Looks like a good point for improvement. 
I believe initial limitation in `255 symbols length` exists from the ancient times and need to be fixed.

> Before Mysql version 5.0.3 Varchar datatype can store 255 character, but from 5.0.3 it can be store 65,535 characters. BUT it has a limitation of maximum row size of 65,535 bytes. It means including all columns it must not be more than 65,535 bytes

Reply:
>not sure that 255 symbols would be enough for us
I'm not sure I'm getting the point; I used a `64k` size, why do you refer to 255?

Reply:
@aleron75 
First of all,  I think 64k is too much space for this field. 
Another thing is that currently, this length would result in creation MySQL  data type `text ` because of 
```php
elseif ($length > 255 && $length <= 65536) {
                    $cType = $ddlType == Table::TYPE_TEXT ? 'text' : 'blob';
                } 
```
we definetly don't want to use `text` on the level of MySQL. Because this is not efficient from the MySQL point of view. 
The most appropriate MySQL data type for our use-case is VARCHAR (4096)
But currently, there is no possibility to create one, as we have limitations on the framework level. Magento creates varchars up to 255 symbols length only.